src/PostgREST/Types.hs

```diff
@@ -56,11 +56,26 @@ decodeContentType ct = case BS.takeWhile (/= BS.c2w ';') ct of
   "*/*"                               -> CTAny
   ct'                                 -> CTOther ct'
 
+-- | A SQL query that can be executed independently
+type SqlQuery = Text
+
+-- | A part of a SQL query that cannot be executed independently
+type SqlFragment = Text
+
 data PreferResolution = MergeDuplicates | IgnoreDuplicates deriving Eq
 instance Show PreferResolution where
   show MergeDuplicates  = "resolution=merge-duplicates"
   show IgnoreDuplicates = "resolution=ignore-duplicates"
 
+data PreferParameters =
+  -- | Pass all parameters as a single json object to a stored procedure
+  SingleObject |
```

Comment:
sorry another nitpick: why not the as follows?
reason: readability
``` 
data PreferParameters
  = SingleObject -- Pass all parameters as a single json object to a stored procedure
  | MultipleObjects -- Pass an array of json objects as params to a stored procedure
   deriving Eq

instance Show PreferParameters where
  show SingleObject    = "params=single-object"
  show MultipleObjects = "params=multiple-objects"
```

Reply:
I came across the conventions only after posting the comment, again thanks for the link, another thing I learned :)